src/Hooks/GutenbergCleanup.php

```diff
@@ -0,0 +1,117 @@
+<?php
+
+namespace WPML\PB\Elementor\Hooks;
+
+use WPML\FP\Fns;
+use WPML\FP\Maybe;
+use WPML\FP\Relation;
+use WPML\LIB\WP\Post;
+use WPML\PB\Elementor\DataConvert;
+use WPML_Elementor_Data_Settings;
+use function WPML\FP\curryN;
+
+class GutenbergCleanup implements \IWPML_Backend_Action, \IWPML_Frontend_Action {
+
+	public function add_hooks() {
+		add_filter(
+			'update_post_metadata',
+			Fns::withoutRecursion( Fns::identity(), [ $this, 'removeGutenbergFootprint' ] ),
+			10,
+			4
+		);
+	}
+
+	/**
+	 * If we detect Gutenberg footprint in the Elementor data,
+	 * we'll remove it and delete the Gutenberg string package.
+	 *
+	 * @param null|bool $check
+	 * @param int       $postId
+	 * @param string    $metaKey
+	 * @param string    $metaValue
+	 *
+	 * @return mixed
+	 */
+	public function removeGutenbergFootprint( $check, $postId, $metaKey, $metaValue ) {
+		if (
+			WPML_Elementor_Data_Settings::META_KEY_DATA === $metaKey
+			&& WPML_Elementor_Data_Settings::is_edited_with_elementor( $postId )
+		) {
+			$data = DataConvert::unserialize( $metaValue );
```

Comment:
@strategio I think you can now move this into the Maybe:

```php
			return Maybe::of( $metaValue )
				->map( [ DataConvert::class, 'unserialize' ] )
				->map( [ $this, 'removeBlockMetaInEditorWidget' ] )
				->filter( $ifDifferentFrom( $data ) )
				->map( [ DataConvert::class, 'serialize' ] )
				->map( $update( $postId ) )
				->getOrElse( $check );
```

Reply:
Unfortunately not @brucepearson, we need `$data` for the `$ifDifferentFrom` function. Or perhaps you have something to suggest?

Reply:
@strategio I guess you can't because we need `$data` in the filter function

Reply:
@strategio Actually, maybe you could do:

```
return Maybe::of( $metaValue )
				->map( [ DataConvert::class, 'unserialize' ] )
				->map( [ $this, 'removeBlockMetaInEditorWidget' ] )
				->map( [ DataConvert::class, 'serialize' ] )
				->filter( pipe( Relation::equals( $metaValue ), Fns::not() ) )
				->map( $update( $postId ) )
				->getOrElse( $check );

Reply:
That's completely true @brucepearson, why would we make it complicated when it could be so simple :)